pkg/proxy/proxy.go

```diff
@@ -228,67 +229,100 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			if p.isNotSupported(vin) {
 				p.forwardRequest(acct.Host, w, req)
 			} else {
-				p.handleVehicleCommand(acct, w, req, command, vin)
+				if err := p.handleVehicleCommand(acct, w, req, command, vin); err == vehicle.ErrCommandUseRESTAPI {
+					p.forwardRequest(acct.Host, w, req)
+				}
 			}
 			return
 		}
 	}
 	p.forwardRequest(acct.Host, w, req)
 }
 
-func (p *Proxy) handleVehicleCommand(acct *account.Account, w http.ResponseWriter, req *http.Request, command string, vin string) {
+func (p *Proxy) handleVehicleCommand(acct *account.Account, w http.ResponseWriter, req *http.Request, command, vin string) error {
 	ctx, cancel := context.WithTimeout(context.Background(), p.Timeout)
 	defer cancel()
 
 	// Serialize commands sent to a specific VIN to avoid some complexities associated with sharing
 	// the vehicle.Vehicle object. VCSEC commands fail if they arrive out of order, anyway.
 	if err := p.lockVIN(ctx, vin); err != nil {
 		writeJSONError(w, http.StatusServiceUnavailable, err)
-		return
+		return err
 	}
 	defer p.unlockVIN(vin)
 
-	log.Debug("Executing %s on %s", command, vin)
-	if req.Method != http.MethodPost {
-		writeJSONError(w, http.StatusMethodNotAllowed, nil)
-		return
-	}
-	car, err := acct.GetVehicle(ctx, vin, p.commandKey, p.sessions)
-	if err != nil || car == nil {
-		writeJSONError(w, http.StatusInternalServerError, err)
-		return
+	car, commandToExecuteFunc, err := p.loadVehicleAndCommandFromRequest(ctx, acct, w, req, command, vin)
+	if err != nil {
+		return err
 	}
 
 	if err := car.Connect(ctx); err != nil {
 		writeJSONError(w, http.StatusInternalServerError, err)
-		return
+		return err
 	}
 	defer car.Disconnect()
 
 	if err := car.StartSession(ctx, nil); err == protocol.ErrProtocolNotSupported {
 		p.markUnsupportedVIN(vin)
 		p.forwardRequest(acct.Host, w, req)
```

Comment:
If `handleVehicleCommand` passes down errors, then `forwardRequest` probably should as well.

Reply:
I look into it, not sure it make sense. I don't see any utilities for error from forwardRequest